lib/collection/src/collection.rs

```diff
@@ -642,38 +643,128 @@ impl Collection {
             .await
     }
 
-    async fn _search(
+    pub async fn search_batch(
         &self,
-        request: SearchRequest,
+        request: SearchRequestBatch,
         search_runtime_handle: &Handle,
         shard_selection: Option<ShardId>,
-    ) -> CollectionResult<Vec<ScoredPoint>> {
+    ) -> CollectionResult<Vec<Vec<ScoredPoint>>> {
+        // A factor which determines if we need to use the 2-step search or not
+        // Should be adjusted based on usage statistics.
+        const PAYLOAD_TRANSFERS_FACTOR_THRESHOLD: usize = 10;
```

Comment:
the logic is adapted from the regular `search` taking in consideration the batching aspects